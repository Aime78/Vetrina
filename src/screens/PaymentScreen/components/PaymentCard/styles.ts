import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    marginBottom: 20,
    backgroundColor: 'white',
    borderRadius: 10,
    width: 370,
    height: 150,
    padding: 20,
    justifyContent: 'space-between',
  },
  shadow: {
    shadowColor: '#171717',
    shadowOffset: {width: -2, height: 4},
    shadowOpacity: 0.3,
    shadowRadius: 3,
  },
  containerHeader: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  text: {
    fontSize: 17,
    fontWeight: '600',
    color: '#103B66',
  },
});

export default styles;
