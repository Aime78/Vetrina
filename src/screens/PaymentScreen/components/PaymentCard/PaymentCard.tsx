import {View, Text, Image} from 'react-native';
import styles from './styles';

type PaymentCardProps = {
  icon: React.ReactNode;
  name: string;
  status: React.ReactNode;
};
const PaymentCard = ({icon, name, status}: PaymentCardProps) => {
  return (
    <View style={[styles.container, styles.shadow]}>
      <View style={styles.containerHeader}>
        <View>{icon}</View>
        <View>{status}</View>
      </View>
      <Text style={styles.text}>{name}</Text>
    </View>
  );
};

export default PaymentCard;
