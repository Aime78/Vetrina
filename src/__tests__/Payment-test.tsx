import React from 'react';
import renderer from 'react-test-renderer';
import PaymentCard from 'screens/PaymentScreen/components/PaymentCard/PaymentCard';
import methods from 'screens/PaymentScreen/cardData';

it('renders the all products button of the new product screen appropriately', () => {
  const tree = renderer
    .create(
      <PaymentCard
        icon={methods[0].method}
        name={methods[0].methodName}
        status={methods[0].status}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
