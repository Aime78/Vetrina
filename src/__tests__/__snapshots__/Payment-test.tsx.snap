// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the all products button of the new product screen appropriately 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": 150,
        "justifyContent": "space-between",
        "marginBottom": 20,
        "padding": 20,
        "width": 370,
      },
      {
        "shadowColor": "#171717",
        "shadowOffset": {
          "height": 4,
          "width": -2,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
      },
    ]
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View>
      <Image
        source={
          {
            "testUri": "../../../src/assets/cash.png",
          }
        }
      />
    </View>
    <View>
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderRadius": 4,
              "height": 32,
              "justifyContent": "center",
              "width": 80,
            },
            {
              "backgroundColor": "#E0FFF6",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontSize": 14,
                "fontWeight": "400",
              },
              {
                "color": "#00C48C",
              },
            ]
          }
        >
          Disable
        </Text>
      </View>
    </View>
  </View>
  <Text
    style={
      {
        "color": "#103B66",
        "fontSize": 17,
        "fontWeight": "600",
      }
    }
  >
    Cash
  </Text>
</View>
`;
